In [1]:
import pandas as pd

# Load the data
file_path = r"C:\Users\neba\Desktop\Insurance-Analytics\Data\cleaned_insurance_data.csv"
insurance_data = pd.read_csv(file_path)

# Display the first few rows of the dataset
insurance_data.head()

C:\Users\neba\AppData\Local\Temp\ipykernel_10404\2090046736.py:5: DtypeWarning: Columns (32) have mixed types. Specify dtype option on import or set low_memory=False.
  insurance_data = pd.read_csv(file_path)


UnderwrittenCoverID  PolicyID TransactionMonth  IsVATRegistered  \
0               145249     12827       2015-03-01             True   
1               145249     12827       2015-05-01             True   
2               145249     12827       2015-07-01             True   
3               145255     12827       2015-05-01             True   
4               145255     12827       2015-07-01             True   

  Citizenship          LegalType Title Language                 Bank  \
0              Close Corporation    Mr  English  First National Bank   
1              Close Corporation    Mr  English  First National Bank   
2              Close Corporation    Mr  English  First National Bank   
3              Close Corporation    Mr  English  First National Bank   
4              Close Corporation    Mr  English  First National Bank   

       AccountType  ... CoverCategory   CoverType            CoverGroup  \
0  Current account  ...    Windscreen  Windscreen  Comprehensive - Taxi   
1  Current account  ...    Windscreen  Windscreen  Comprehensive - Taxi   
2  Current account  ...    Windscreen  Windscreen  Comprehensive - Taxi   
3  Current account  ...    Own damage  Own Damage  Comprehensive - Taxi   
4  Current account  ...    Own damage  Own Damage  Comprehensive - Taxi   

               Section                          Product StatutoryClass  \
0  Motor Comprehensive  Mobility Metered Taxis: Monthly     Commercial   
1  Motor Comprehensive  Mobility Metered Taxis: Monthly     Commercial   
2  Motor Comprehensive  Mobility Metered Taxis: Monthly     Commercial   
3  Motor Comprehensive  Mobility Metered Taxis: Monthly     Commercial   
4  Motor Comprehensive  Mobility Metered Taxis: Monthly     Commercial   

  StatutoryRiskType TotalPremium  TotalClaims VehicleAge  
0     IFRS Constant    21.929825          0.0       11.0  
1     IFRS Constant    21.929825          0.0       11.0  
2     IFRS Constant     0.000000          0.0       11.0  
3     IFRS Constant   512.848070          0.0       11.0  
4     IFRS Constant     0.000000          0.0       11.0  

[5 rows x 48 columns]

Risk Differences Across Provinces
Null Hypothesis (H₀): There are no risk differences across provinces.
Alternative Hypothesis (H₁): There are significant risk differences across provinces.
KPI: TotalClaims or TotalPremium
Test: ANOVA (if provinces > 2), T-test (if comparing two provinces)

In [2]:
import scipy.stats as stats

# Group data by province and perform ANOVA
province_groups = [group['TotalClaims'].dropna() for _, group in insurance_data.groupby('Province')]

# Perform ANOVA test
f_stat, p_value = stats.f_oneway(*province_groups)

print("ANOVA Test for Risk Differences Across Provinces")
print(f"F-statistic: {f_stat}, p-value: {p_value}")

if p_value < 0.05:
    print("Reject Null Hypothesis: There are significant risk differences across provinces.")
else:
    print("Fail to Reject Null Hypothesis: No significant risk differences across provinces.")

ANOVA Test for Risk Differences Across Provinces
F-statistic: 5.853051061976046, p-value: 1.6569173744506565e-07
Reject Null Hypothesis: There are significant risk differences across provinces.


Risk Differences Between Zip Codes
Null Hypothesis (H₀): There are no risk differences between zip codes.
Alternative Hypothesis (H₁): There are significant risk differences between zip codes.
KPI: TotalClaims or TotalPremium
Test: ANOVA (for multiple zip codes), T-test (for two zip codes)

In [6]:
# Group data by PostalCode and perform ANOVA
postal_groups = [group['TotalClaims'].dropna() for _, group in insurance_data.groupby('PostalCode')]

# Perform ANOVA test
f_stat, p_value = stats.f_oneway(*postal_groups)

print("ANOVA Test for Risk Differences Across Postal Codes")
print(f"F-statistic: {f_stat}, p-value: {p_value}")

if p_value < 0.05:
    print("Reject Null Hypothesis: There are significant risk differences between postal codes.")
else:
    print("Fail to Reject Null Hypothesis: No significant risk differences between postal codes.")

ANOVA Test for Risk Differences Across Postal Codes
F-statistic: 0.9419190994382765, p-value: 0.8908852586661922
Fail to Reject Null Hypothesis: No significant risk differences between postal codes.


In [7]:
# Group data by PostalCode and perform ANOVA
postal_profit_groups = [group['ProfitMargin'].dropna() for _, group in insurance_data.groupby('PostalCode')]

# Perform ANOVA test
f_stat, p_value = stats.f_oneway(*postal_profit_groups)

print("ANOVA Test for Profit Margin Differences Across Postal Codes")
print(f"F-statistic: {f_stat}, p-value: {p_value}")

if p_value < 0.05:
    print("Reject Null Hypothesis: There are significant profit margin differences between postal codes.")
else:
    print("Fail to Reject Null Hypothesis: No significant profit margin differences between postal codes.")

ANOVA Test for Profit Margin Differences Across Postal Codes
F-statistic: 0.8704926765547479, p-value: 0.9977282177272306
Fail to Reject Null Hypothesis: No significant profit margin differences between postal codes.


Risk Differences Between Women and Men
Null Hypothesis (H₀): There are no significant risk differences between Women and Men.
Alternative Hypothesis (H₁): There are significant risk differences between Women and Men.
KPI: TotalClaims
Test: T-test (as gender has only two groups)

In [8]:
# Split data into male and female groups
male_group = insurance_data[insurance_data['Title'].str.contains('Mr', na=False)]['TotalClaims']
female_group = insurance_data[insurance_data['Title'].str.contains('Mrs|Ms|Miss', na=False)]['TotalClaims']

# Perform T-test
t_stat, p_value = stats.ttest_ind(male_group.dropna(), female_group.dropna())

print("T-test for Risk Differences Between Men and Women")
print(f"T-statistic: {t_stat}, p-value: {p_value}")

if p_value < 0.05:
    print("Reject Null Hypothesis: There are significant risk differences between Women and Men.")
else:
    print("Fail to Reject Null Hypothesis: No significant risk differences between Women and Men.")

T-test for Risk Differences Between Men and Women
T-statistic: 2.0371487157445594, p-value: 0.041635392696217785
Reject Null Hypothesis: There are significant risk differences between Women and Men.


Here’s your report based on the statistical outcomes:  

---

# **A/B Hypothesis Testing Report**  

## **1. Risk Differences Across Provinces**  
- **Null Hypothesis (H₀):** There are no risk differences across provinces.  
- **Alternative Hypothesis (H₁):** There are significant risk differences across provinces.  
- **KPI:** TotalClaims or TotalPremium  
- **Test Used:** ANOVA  

### **Results**  
- **F-statistic:** 5.85  
- **p-value:** 1.66e-07  

### **Interpretation**  
Since the p-value is significantly below the 0.05 threshold, we **reject the null hypothesis**. This indicates that risk levels (as measured by claims or premiums) vary significantly across different provinces.  

### **Business Implications**  
- Risk-based pricing should consider provincial differences, as certain provinces may have higher claim frequencies or severities.  
- Marketing and risk assessment strategies should be tailored based on the province to optimize premiums and minimize loss ratios.  

---

## **2. Risk Differences Between Postal Codes**  
- **Null Hypothesis (H₀):** There are no risk differences between postal codes.  
- **Alternative Hypothesis (H₁):** There are significant risk differences between postal codes.  
- **Test Used:** ANOVA  

### **Results**  
- **F-statistic:** 0.94  
- **p-value:** 0.89  

### **Interpretation**  
Since the p-value is much greater than 0.05, we **fail to reject the null hypothesis**. This suggests that risk levels do not vary significantly between postal codes.  

### **Business Implications**  
- Postal codes alone may not be a strong predictor of risk.  
- Other geographic or demographic factors should be explored for more granular risk segmentation.  

---

## **3. Profit Margin Differences Across Postal Codes**  
- **Null Hypothesis (H₀):** There are no significant profit margin differences between postal codes.  
- **Alternative Hypothesis (H₁):** There are significant profit margin differences between postal codes.  
- **Test Used:** ANOVA  

### **Results**  
- **F-statistic:** 0.87  
- **p-value:** 0.998  

### **Interpretation**  
Since the p-value is significantly greater than 0.05, we **fail to reject the null hypothesis**. This means there is no statistical evidence to suggest that profit margins vary significantly by postal code.  

### **Business Implications**  
- Profitability is likely influenced by other variables such as customer profile, vehicle type, or policy coverage rather than location alone.  
- Pricing strategies should focus on broader risk factors rather than postal code segmentation alone.  

---

## **4. Risk Differences Between Men and Women**  
- **Null Hypothesis (H₀):** There are no significant risk differences between men and women.  
- **Alternative Hypothesis (H₁):** There are significant risk differences between men and women.  
- **Test Used:** T-test  

### **Results**  
- **T-statistic:** 2.04  
- **p-value:** 0.0416  

### **Interpretation**  
Since the p-value is less than 0.05, we **reject the null hypothesis**, indicating that there are statistically significant risk differences between men and women.  

### **Business Implications**  
- Gender-based risk assessment may be necessary for optimizing premium calculations.  
- Regulatory considerations should be taken into account when adjusting premiums based on gender.  
- Further investigation into behavioral factors contributing to risk disparities could improve underwriting policies.  

---

## **Conclusion**  
- **Significant risk differences exist across provinces and between genders.** These factors should be incorporated into pricing and underwriting strategies.  
- **No significant risk differences were found between postal codes.** Location at the postal code level may not be a useful risk predictor.  
- **Profit margins do not significantly vary by postal code.** Other variables should be analyzed to optimize profitability.  

This analysis provides insights that can drive **pricing optimization, targeted marketing, and risk assessment improvements** in the insurance strategy. 🚀  